### ISSUE TITLE
Intentionally empty repo?

### ISSUE THREAD

**Issue by dguo** [purpliminal/rust-dotenv#1]
Hi, dotenv's [Cargo.toml](https://docs.rs/crate/dotenv/0.13.0/source/Cargo.toml) refers to this repo, but it's empty. Is that intentional?

**Comment by TuringZhu** [comment]
@dguo the right repo: [bkeepers/dotenv](https://github.com/bkeepers/dotenv), DO NOT click HomePage

**Comment by dguo** [comment]
That repo is for a Ruby implementation of dotenv. I'm looking for the Rust implementation.

I don't know what you mean by "DO NOT click HomePage."

**Comment by lweberk** [comment]
Same here... looking for the repository for this crate. Any pointers to where it went?

**Comment by nlfiedler** [comment]
Yeah, that's bizarre. Someone went out of their way to remove everything. There is an old version of the code here https://github.com/cmr/rust-dotenv in case that helps any.

**Comment by azriel91** [comment]
https://github.com/apiraino/rust-dotenv is the most up to date, based on the [network graph](https://github.com/cmr/rust-dotenv/network). Its [Cargo.toml](https://github.com/apiraino/rust-dotenv/blob/5cfd95025b8e38de76937cfca525562115396ae2/Cargo.toml#L16) points to apiraino/rust-dotenv, so it should be fixed on the next publish.

**Comment by lweberk** [comment]
I've been meaning to write this for a couple of days now... sorry for not having got to it. 

Sean Griffin (co-author of this lib) was so kind to point me to the crates.io API to get the latest available version as source code, which currently is 0.13.0. It would be lovely if someone could spawn or update a fork of this repository with it to take over the head of the network.

`curl -L crates.io/api/v1/crates/dotenv/0.13.0/download --output dotenv_0.13.0.tar.gz`

**Comment by dguo** [comment]
We could fork it, but we'd also need to get added as owners on crates.io. Otherwise, we wouldn't be able to publish new versions of the crate.

@sgrif, are you looking for new maintainers?

**Comment by Dylan-DPC-zz** [comment]
Yes, the repo is intentionally blank. 

Since the original author of dotenv (the real user named purplinimal) deleted the repo, it left a chance for a malicious user to create a github user with the same name and then create a repo named dotenv. Since crates.io will point to purplinimal/dotenv (which is how you got here) you won't realise the difference.

This user and repo was created just as a placeholder so that nobody misuses the repo link. It will be deleted once we restore the contents of dotenv in another repo (or it is moved to this repo). I should have left a note here regarding this)

**Comment by dguo** [comment]
Just curious, what's holding up restoring the contents in another repo? If you do that, update the link in `Cargo.toml`, and publish a new version, then crates.io will point to the correct repo, won't it?

I know finding time for open source work isn't always easy. It just seems like the time it takes to create a new user and blank repo is about the same time as it takes to restore the repo under an active user's account and update the crates.io metadata.

Thank you for the explanation.

**Comment by jimmycuadra** [comment]
I'm also wondering where the source code lives now. I'm not able to build dotenv 0.10.1. I get this error:

```
error[E0659]: `error_chain` is ambiguous (derive helper attribute vs any other name)
  --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/dotenv-0.10.1/src/lib.rs:23:40
   |
23 | #[cfg_attr(not(feature = "backtrace"), error_chain(backtrace = "false"))]
   |                                        ^^^^^^^^^^^ ambiguous name
   |
note: `error_chain` could refer to the derive helper attribute defined here
  --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/dotenv-0.10.1/src/lib.rs:22:17
   |
22 | #[derive(Debug, error_chain)]
   |                 ^^^^^^^^^^^
note: `error_chain` could also refer to the derive macro imported here
  --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/dotenv-0.10.1/src/lib.rs:10:1
   |
10 | #[macro_use]
   | ^^^^^^^^^^^^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0659`.
```

But I don't know where to report it since this repo is empty and it hasn't been restored somewhere else. It's been two months since the last activity on this issue. Anyone know what's going on?

**Comment by ghost** [comment]
@jimmycuadra I've started taking over maintainership of dotenv. You can find the repo at https://github.com/dotenv-rs/dotenv.

**Comment by Dylan-DPC-zz** [comment]
@dgriffen great. Would be worth adding in the readme that this (your repo) isn't the official dotenv but a maintained fork. Thanks

**Comment by ghost** [comment]
@Dylan-DPC it _will_ be published as the official dotenv. The dotenv-rs org has permissions to publish crates.

**Comment by Dylan-DPC-zz** [comment]
@dgriffen okay that's great. Let me know once there is a new release on crates.io and it reflects your repo, then i'll delete this repo.

**Comment by ghost** [comment]
@Dylan-DPC the new release is up. The only repo we couldnt get ahold of was `dotenv_codegen_impl` since there was only one listed owner. We renamed that crate since it wasnt a public dependency.

**Comment by ghost** [comment]
One thing I've noticed, this account will have to remain around since it is still listed as an owner on the crates. @Dylan-DPC would it be possible for you to add @sgrif as an owner of `dotenv_codegen_impl`?

**Comment by sgrif** [comment]
They wouldn't be able to, they're not an owner of the crate

**Comment by ghost** [comment]
Oh, for some reason I thought that it was tied to the github username. I'm guessing is some more unique identifier?

**Comment by Dylan-DPC-zz** [comment]
@dgriffen crates.io still considers the original name and my account as 2 different accounts ( a good thing) so I can't add you as a co-owner. Either Sean has permissions or you could start maintaining a fork and swap it for that (worst-case)
